third_party/terraform/resources/resource_storage_bucket.go

```diff
@@ -755,89 +768,162 @@ func flattenBucketLifecycleRuleCondition(condition *storage.BucketLifecycleRuleC
-					condition := conditions[0].(map[string]interface{})
+	if v, ok := rule["action"]; ok {
+		action, err := expandStorageBucketLifecycleRuleAction(v)
+		if err != nil {
+			return nil, err
+		}
+		transformed.Action = action
+	} else {
+		return nil, fmt.Errorf("exactly one action is required for lifecycle_rule")
+	}
 
-					target_lifecycle_rule.Condition = &storage.BucketLifecycleRuleCondition{}
+	if v, ok := rule["condition"]; ok {
+		cond, err := expandStorageBucketLifecycleRuleCondition(v)
+		if err != nil {
+			return nil, err
+		}
+		transformed.Condition = cond
+	}
 
-					if v, ok := condition["age"]; ok {
-						target_lifecycle_rule.Condition.Age = int64(v.(int))
-					}
+	return transformed, nil
+}
 
-					if v, ok := condition["created_before"]; ok {
-						target_lifecycle_rule.Condition.CreatedBefore = v.(string)
-					}
+func expandStorageBucketLifecycleRuleAction(v interface{}) (*storage.BucketLifecycleRuleAction, error) {
+	if v == nil {
+		return nil, fmt.Errorf("exactly one action is required for lifecycle_rule")
+	}
 
-					if v, ok := condition["is_live"]; ok {
-						target_lifecycle_rule.Condition.IsLive = googleapi.Bool(v.(bool))
-					}
+	actions := v.(*schema.Set).List()
+	if len(actions) != 1 {
+		return nil, fmt.Errorf("exactly one action is required for lifecycle_rule")
+	}
 
-					if v, ok := condition["matches_storage_class"]; ok {
-						matches_storage_classes := v.([]interface{})
+	action := actions[0].(map[string]interface{})
+	transformed := &storage.BucketLifecycleRuleAction{}
 
-						target_matches_storage_classes := make([]string, 0, len(matches_storage_classes))
+	if v, ok := action["type"]; ok {
+		transformed.Type = v.(string)
+	}
 
-						for _, v := range matches_storage_classes {
-							target_matches_storage_classes = append(target_matches_storage_classes, v.(string))
-						}
+	if v, ok := action["storage_class"]; ok {
+		transformed.StorageClass = v.(string)
+	}
 
-						target_lifecycle_rule.Condition.MatchesStorageClass = target_matches_storage_classes
-					}
+	return transformed, nil
+}
 
-					if v, ok := condition["num_newer_versions"]; ok {
-						target_lifecycle_rule.Condition.NumNewerVersions = int64(v.(int))
-					}
-				} else {
-					return fmt.Errorf("Exactly one condition is required")
-				}
+func expandStorageBucketLifecycleRuleCondition(v interface{}) (*storage.BucketLifecycleRuleCondition, error) {
+	if v == nil {
+		return nil, nil
+	}
+	conditions := v.(*schema.Set).List()
+	if len(conditions) != 1 {
+		return nil, fmt.Errorf("One and only one condition can be provided per lifecycle_rule")
+	}
+
+	condition := conditions[0].(map[string]interface{})
+	transformed := &storage.BucketLifecycleRuleCondition{}
+
+	if v, ok := condition["age"]; ok {
+		transformed.Age = int64(v.(int))
+	}
+
+	if v, ok := condition["created_before"]; ok {
+		transformed.CreatedBefore = v.(string)
+	}
+
+	// Because TF schema, withStateOk will always be true.
```

Comment:
If it's always true why is wrapped in an `if` at all?

Reply:
I don't know if the schema behavior will change? I'll add a just in case.